aptos-move/aptos-release-builder/src/components/framework.rs

```diff
@@ -35,23 +43,41 @@ pub fn generate_upgrade_proposals(is_testnet: bool) -> Result<Vec<(String, Strin
             .unwrap()
             .to_string();
 
+        let mut args = vec![
+            "run",
+            "--bin",
+            "aptos",
+            "--",
+            "governance",
+            "generate-upgrade-proposal",
+            "--account",
+            publish_addr,
+            "--output",
+            move_script_path.to_str().unwrap(),
+            "--package-dir",
+            package_path.to_str().unwrap(),
+        ];
+
+        if is_testnet {
+            args.push("--testnet");
+        }
+
+        let mut _execution_hash: String = "".to_owned();
+        if !next_execution_hash.is_empty() {
+            args.push("--next-execution-hash");
+            if result.is_empty() {
```

Comment:
I don't quite follow the flow here. Can you add some comments to clarify?

Reply:
if result.is_empty(), it means that this is the first framework file, so its `next_execution_hash` would be whatever is being passed in (e.g. a `version`/`feature_flag` file's hash etc.). 

if result is not empty, it means that the next execution hash should be the hash of `result.last()`.

I'll add more comments!

Reply:
What do you mean by the first framework file? How would the command work if we need to chain framework upgrades with on chain flag changes?

Reply:
ah I guess we don't support custom on-chain flag changes yet (unless it's part of `feature_flag`?)

let's say we are going to generate these files: 
```
0-move-stdlib.move	2-aptos-framework.move	4-gas-schedule.move	6-features.move
1-aptos-stdlib.move	3-aptos-token.move	5-version.move		7-consensus-config.move
```
the first framework file being generated is `3-aptos-token.move`. it's using the `next_execution_hash` being passed in (so in this case, the hash of `4-gas-schedule.move`. 
the second framework file would be `2-aptos-framework.move`, and it's using the hash of `3-aptos-token.move` (which would be `result.last()`). 
I realized that the "first framework file being generated" can sound confusing, because we are reversing it in the end so it becomes the last framework file.

Reply:
I see. That makes sense then